src/components/Viewer.js

```diff
@@ -60,69 +62,111 @@ function Viewer ({usfm}) {
           }
         });
 
-        filteredVariantObjects.push(currentVariantObject);
+        if (currentVariantObject.readings.length > 0)
+        {
+          filteredVariantObjects.push(currentVariantObject);
+        }
       }
     );
 
-    return variantObjects;
+    return filteredVariantObjects;
+  };
+
+  Array.prototype.insert = function ( index, item ) {
+    this.splice( index, 0, item );
+  };
+  
+  const tokenizer = (text) => {
+    return tokenize({
+      text: text,
+      greedy: true,
+      normalize: true,
+    });
+  };
+
+  const normalizeString = (string) => {
+    const normalized = tokenizer(string).join(' ').replace('ʼ', '').replace('’', '');
+    return normalized;
+  };
+
+  const tagVariantsInVerseObjects = (currentVerseObjects, filteredVariantObjects) =>
+  {
+      if (currentVerseObjects)
+      {
+        filteredVariantObjects.forEach(variant =>{
+          const variantBaseTextSplit = variant.baseText.split(" ").filter(txt => txt.trim().length > 0);
+
+          currentVerseObjects.verseObjects.forEach(vo => {
+            if (normalizeString(vo.text) === normalizeString(variantBaseTextSplit[0]))
+            {
+              vo.text = '⸢' + vo.text;
+            }
+            if (normalizeString(vo.text) === normalizeString(variantBaseTextSplit[variantBaseTextSplit.length-1]))
+            {
+              vo.text = vo.text + '⸣';
+            }
+          });
+        });
+      }
+
+      return currentVerseObjects;
   };
   
   const verses = Object.keys(chapters[chapterKey]).filter(key => key.match(/^\d+$/)).map(
     (currentVerseKey) => {
-      let currentVerseObjects = [];
-      currentVerseObjects[1] = chapters[chapterKey][currentVerseKey];
-
+      //let currentVerseObjects = [];
+      //currentVerseObjects[1] = chapters[chapterKey][currentVerseKey];
+      
       const filteredVariantObjects = filterApparatusData(chapterKey, currentVerseKey);
+      const currentVerseObjectsArrayContents = tagVariantsInVerseObjects(chapters[chapterKey][currentVerseKey], filteredVariantObjects);
+      const currentVerseObjects = JSON.parse('{"' + parseInt(currentVerseKey) + '": ' + JSON.stringify(currentVerseObjectsArrayContents) + '}');
 
       //const apparatusData = JSON.stringify(filterApparatusData(chapterKey, currentVerseKey), null, 4)
```

Comment:
Shall we remove this now?

Reply:
@richmahn  remove what please?

Reply:
I think he's referring to the commented code